typing/typecore.ml

```diff
@@ -6859,6 +6918,15 @@ let report_error ~loc env = function
         "@[<2>%s:@ %a@]"
         "This type does not bind all existentials in the constructor"
         (Style.as_inline_code pp_type) (ids, ty)
+  | Bind_existential (reason, id, ty) ->
+      let reason = match reason with
+      | Bind_already_bound -> "already bound existential"
+      | Bind_not_in_scope -> "previously defined existential"
+      | Bind_non_locally_abstract -> "non-locally-abstract type"
```

Comment:
I think that this error message could be more user-friendly. This corresponds to the common case where the user tries to bind an "old" type. "non-locally-abstract" is a correct description but it will be confusing for users.

For example I wouldn't mind an error message saying something like:

> The local name ... can only be given to an existential variable bound to this GADT constructor. The type annotation tries to bind it to the name ... defined outside this constructor.

Reply:
Done as ssugested